edge_computing/day_2_core_cnf_clusters/upgrading/telco-upgrade-completing-the-upgrade.adoc

```diff
@@ -0,0 +1,86 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="telco-upgrade-completing-the-upgrade"]
+= Completing the telco core CNF cluster upgrade
+include::_attributes/common-attributes.adoc[]
+:context: completing-the-upgrade
+
+toc::[]
+
+Use this information to perform the cluster upgrade and monitor the upgrade through to completion.
+
+include::modules/telco-upgrade-acknowledging-the-upgrade.adoc[leveloffset=+1]
+
+include::modules/telco-upgrade-starting-the-cluster-upgrade.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../../../edge_computing/day_2_core_cnf_clusters/upgrading/telco-upgrade-api.adoc#telco-upgrade-selecting-the-target-release_telco-upgrade-api[Selecting the target release]
+
+include::modules/telco-upgrade-monitoring-the-cluster-upgrade.adoc[leveloffset=+1]
+
+include::modules/telco-upgrade-upgrading-the-olm-operators.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../../../edge_computing/day_2_core_cnf_clusters/upgrading/telco-upgrade-completing-the-upgrade.adoc#telco-upgrade-upgrading-the-worker-nodes_completing-the-upgrade[Upgrading the worker nodes]
+
+include::modules/telco-upgrade-performing-the-second-y-stream-update.adoc[leveloffset=+2]
+
+include::modules/telco-upgrade-acknowledging-the-y-stream-release-upgrade.adoc[leveloffset=+2]
+
+[role="_additional-resources"]
+.Additional resources
+
+* link:https://access.redhat.com/articles/6958395[Preparing to upgrade]
+
+include::modules/telco-upgrade-starting-the-y-stream-control-plane-upgrade.adoc[leveloffset=+1]
+
+include::modules/telco-upgrade-monitoring-second-part-y-upgrade.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../../../edge_computing/day_2_core_cnf_clusters/upgrading/telco-upgrade-completing-the-upgrade.adoc#telco-upgrade-monitoring-the-cluster-upgrade_completing-the-upgrade[Monitoring the cluster upgrade]
+
+include::modules/telco-upgrade-upgrading-all-the-olm-operators.adoc[leveloffset=+1]
+
+[role="_additional-resources"]
+.Additional resources
+
+* xref:../../../edge_computing/day_2_core_cnf_clusters/upgrading/telco-upgrade-completing-the-upgrade.adoc#telco-upgrade-monitoring-the-cluster-upgrade_completing-the-upgrade[Monitoring the cluster upgrade]
+
+include::modules/telco-upgrade-upgrading-the-worker-nodes.adoc[leveloffset=+1]
+
+include::modules/telco-upgrade-verifying-the-health-of-the-newly-upgraded-cluster.adoc[leveloffset=+1]
+
+[id="telco-upgrade-completing-y-stream-cluster-upgrade_{context}"]
+== Completing a y-stream cluster upgrade
```

Comment:
Why is this at the bottom here? This seems like this would have been helpful at the top, when deciding what to follow when doing an EUS-to-EUS, or y-stream, or z-stream update.

Though I still think it might be a better user experience to separate these all out into separate procedures, to that users don't have to make decisions partway through this assembly.

And there isn't one for EUS-to-EUS, but seems like there should be.

Reply:
The parent procedure is the EUS-to-EUS update procedure. I take your point though. I will rework this to provide parity for each of these update options.